Dockerfile.vcsim

```diff
@@ -1,4 +1,48 @@
+# Create a builder container
+FROM golang:latest as build
+WORKDIR /go/src/app
+
+# Create appuser to isolate potential vulnerabilities
+# See https://stackoverflow.com/a/55757473/12429735
+ENV USER=appuser
+ENV UID=10001
+RUN adduser \
+    --disabled-password \
+    --gecos "" \
+    --home "/nonexistent" \
+    --shell "/sbin/nologin" \
+    --no-create-home \
+    --uid "${UID}" \
+    "${USER}"
+
+# Update ca-certificates, for final stage.
+RUN apt-get update && apt-get install -y ca-certificates
```

Comment:
If we pin the image digest there's no need to run `apt-get` which could otherwise lead to non-deterministic builds again.